lib/razorpay/utility.rb

```diff
@@ -0,0 +1,23 @@
+require 'razorpay'
+require 'openssl'
+
+# Helper functions are defined here
+module Razorpay
+  class Utility
+    def self.validate_payment_signature(order_id, payment_id, signature)
+      return validate_signature(signature, order_id, payment_id)
+    end
+
+    private
+
+    def self.validate_signature(signature, *fields)
+      secret = Razorpay.auth[:password]
+
+      data = fields.join('|')
+
+      expected_signature = OpenSSL::HMAC.hexdigest('SHA256', secret, data)
+
+      return expected_signature === signature
```

Comment:
We need a constant time comparison here. Ruby doesn't have one inbuilt [yet](https://bugs.ruby-lang.org/issues/10098), so we can copy the one that [Rack has](http://www.rubydoc.info/github/rack/rack/Rack/Utils.secure_compare)